chore(helm): bump postgresql image tag in helm values

mPyKen opened the pull request:
### SUMMARY
Bump default postgres image tag from [14.6.0-debian-11-r13](https://hub.docker.com/layers/bitnami/postgresql/14.6.0-debian-11-r13/images/sha256-3e03f38b67c1b0035beb09729e5b4d073e428c2a7304692db3b0b3eb89104f4f) to [14.17.0-debian-12-r3](https://hub.docker.com/layers/bitnami/postgresql/14.17.0-debian-12-r3/images/sha256-277def15c4f58098f7f1829c60afc6a52e96ac92796d076dc5d06eecce761c26) in order to have a default image that is compatible with both amd64 and arm64 architecture.
Keeping major version at 14 to avoid breaking any instances without a tag specified.

### ISSUE
Ran into the same issue as https://github.com/apache/superset/issues/24292. Problem is fixed by specifying an image tag with arm support.

### TESTING INSTRUCTIONS
Start in a cluster with arm architecture.

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [x] Has associated issue: https://github.com/apache/superset/issues/24292
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

hainenber reviewed (approved):
Got a failed CI check but it's non-blocking. LGTM and thanks for your contribution!

hainenber commented:
I don't understand GitHub's PR approval sometimes 🥲

Antonio-RiveroMartnez reviewed (commented):
Hey! Thanks for taking care of this, I see you updated the values.yaml but didn't bumped the Chart version, also, you might want to run helm dependency update to pull the new version of Postgres and helm-docs to generate any change needed on the README.

mPyKen commented:
Thanks, updated the chart version and ran `helm dependency update`. I didn't update the actual chart dependencies to avoid hitting any breaking changes. Let me know if you prefer to do that in this PR as well.

mistercrunch reviewed (approved):
@mPyKen knowing we have limitedCI around `helm`, please confirm you tested the helm chart manually

rusackas commented:
Looks like it needs some pre-commit love, too. 
Superset uses Git pre-commit hooks courtesy of [pre-commit](https://pre-commit.com/). To install run the following:
```
pip3 install -r requirements/development.txt
pre-commit install
A series of checks will now run when you make a git commit.
```
Alternatively it is possible to run pre-commit by running pre-commit manually:
```
pre-commit run --all-files
```

mPyKen committed:
Update default postgresql image tag in helm values

mPyKen commented:
Fixed the pre-commit hook errors and rebased the branch.

>@mPyKen knowing we have limitedCI around helm, please confirm you tested the helm chart manually

I tested the original change on my arm instance
